bittide/src/Bittide/Calendar.hs

```diff
@@ -38,40 +43,41 @@ it does not care about the type of its entries, this type depends on the compone
 instantiates the calendar.
 -}
 
+type ValidEntry a repetitionBits = (a, Unsigned repetitionBits)
```

Comment:
```suggestion
data ValidEntry a repetitionBits = ValidEntry
  { -- | docs
    ve_entry :: a
    -- | docs
  , ve_repeat :: Unsigned repetitionBits
  }
```

This rids the code of the meaningless `fst`/`snd` and makes it quite natural to put the docs where they belong.

In the docs the difference between _number of times processed_ and _number of times repeated_ should be emphasized. E.g., upon `ve_repeat` being...

 * 0: the entry is processed 1 time (repeated zero times)
 * 1: the entry is processed 2 times (repeated one time)